app/bundles/LeadBundle/Model/ListModel.php

```diff
@@ -855,23 +857,21 @@ public function removeLead($lead, $lists, $manuallyRemoved = false, $batchProces
             $lists = array($lists);
         }
 
-        $persistLists = $deleteLists = array();
-        foreach ($lists as $listid) {
-            if (!isset($this->leadChangeLists[$listid])) {
+        $persistLists = $deleteLists = $dispatchEvents = array();
```

Comment:
Makes the code harder to parse later and difficult to read. Especially if the lines get long.